pandas/io/formats/style.py

```diff
@@ -3523,15 +3521,48 @@ def css_calc(x, left: float, right: float, align: str):
             "value defining the center line or a callable that returns a float"
         )
 
+    rgbas = None
```

Comment:
don't we have a similar utlity in the plotting routines?

Reply:
had a quick look, this is converting mpl colors to hex codes, which is not the usual direction for plotting.

ignoring the arg validation, though which is specific to the method, this op is just a single loop comprehension: `[mpl_colors.rgb2hex(rgba) for rgba in rgbas]`

Reply:
an alternative is to add a `cmap` arg, which accepts a string or mpl ColorMap, similar to `background_gradient` and `text_gradient`.

Advantage is: i) consistency ii) autoloading a colormap from string rather than user have to create a Colormap instance.

Disadvatange is: having a `color` and a `cmap` arg which is slightly confusing although can b documented away.

?

Reply:
hmm i do like the consisteny of i), can we kill the `color` arg (may need to deprecate)

Reply:
not really because you need differentiate between a color which is a valid css str e.g.  "#ee3ee" or "rgb(10,200,10)" or "salmon" and between a string which is a matplotlib colormap, i.e. "inferno" or "PuBu",

Reply:
do we have this color, cmap api anywhere else?

Reply:
The following use `color`: `highligh_min/max/betwen/quantile/null`, and these use `cmap`: `text/background_gradient`. But the use is specific where either a specific color or a color gradient is useful.
For `bar` either is useful.

Matplot lib also separates the arguments on some functions, e.g. `scatter`, where it has a `c`, `color` and `cmap` argument: https://matplotlib.org/stable/api/_as_gen/matplotlib.axes.Axes.scatter.html?highlight=scatter#matplotlib.axes.Axes.scatter